driver-mapping/src/main/java/com/datastax/driver/mapping/config/PropertyAccessStrategy.java

```diff
@@ -0,0 +1,94 @@
+/*
+ *      Copyright (C) 2012-2015 DataStax Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package com.datastax.driver.mapping.config;
+
+import java.beans.PropertyDescriptor;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+
+/**
+ * A strategy to determine how mapped properties are read and written.
+ */
+public interface PropertyAccessStrategy {
+
+    /**
+     * Returns whether or not the given {@link Field} should
+     * be used when accessing its underlying property.
+     *
+     * @return {@code true} if the given field should be considered, {@code false} otherwise.
+     */
+    boolean isFieldAccessAllowed();
+
+    /**
+     * Returns whether or not the given {@link PropertyDescriptor} should
+     * be used when accessing its underlying property.
+     *
+     * @return {@code true} if the given {@link PropertyDescriptor} should be considered, {@code false} otherwise.
+     */
+    boolean isGetterSetterAccessAllowed();
+
+    /**
+     * Locates a getter method for the given base class and given property.
+     * If this method returns {@code null} then no getter will be used to access the given property.
+     *
+     * @param mappedClass The mapped class; this is necessarily a class annotated with
+     *                  either {@link com.datastax.driver.mapping.annotations.Table @Table} or
+     *                  {@link com.datastax.driver.mapping.annotations.UDT @UDT}.
+     * @param property  The property to locate a getter for; never {@code null}.
+     * @return The getter method for the given base class and given property, or {@code null} if no getter was found.
+     */
+    Method locateGetter(Class<?> mappedClass, PropertyDescriptor property);
```

Comment:
Yes, right, of course. That's why I've been working on yet another version of `PropertyAccessStrategy` that _really_ gives full control on what is a mapped property in the first place, and how to access it. It basically boils down to make everything or almost in `PropertyMapper` public, call it stg like `MappedProperty`, and have the `PropertyAccessStrategy` return something like `Set<MappedProperty<?>> mapProperties(List<Class<?>> classHierarchy)`.

Problem is, the more it goes, the more I think all of this is so over-engineered. I'm not even sure I will push the code I have so far.

And no matter what is the outcome of this PR and what the final API looks like, I think it would be wiser to mark it as `@Beta` – otherwise I fear we will regret some, if not all, of the choices made here.